src/databricks/sql/client.py

```diff
@@ -796,13 +797,15 @@ def execute(
             cursor=self,
             use_cloud_fetch=self.connection.use_cloud_fetch,
             parameters=prepared_params,
+            async_op=async_op,
         )
         self.active_result_set = ResultSet(
```

Comment:
The result set that is returned over here is empty and does not have any data.

Reply:
I know, but this will make the code confusing and I do not think it is is necessary.

Reply:
I did this to keep the same logical flow for both execute_async and execute. Like in execute the active_result_set has data and in execute_async since there is no data so it is none. Once data is available the active_result_set will again have data, so logically I felt it made sense

Reply:
the return result comes from the returned value from execute_command in the sync flow, which means it is not ready until the sync completes, this is why I said it is confusing as it should be set in the completion of the async operation, this is standard practice/ways for most of the async code (`on_complete = (result) => { setResult(result) `)

Reply:
I got it what Jacky meant here. This is about confusing the users regarding the interface. We can keep the logic internally same, but don't need to keep the interface same for async and sync. For async what matters is the operationHandle. We can have different interface for both, but internally can reuse the code if possible.

Reply:
@gopalldb @jackyhu-db I have changed the code, based on these suggestions.